src/router.rs

```diff
@@ -198,9 +200,19 @@ mod tests {
     use futures::{executor::block_on, future::FutureObj};
 
     use super::*;
-    use crate::{body::Body, AppData, Request, Response};
+    use crate::{
+        body::Body,
+        middleware::{RequestContext, ResponseContext},
+        AppData, Response,
+    };
+
+    fn passthrough_middleware<Data: Clone + Send>(
+        ctx: RequestContext<Data>,
+    ) -> FutureObj<ResponseContext<Data>> {
+        FutureObj::new(Box::new(ctx.next()))
```

Comment:
nit: no need to make a new `FutureObj` here -- I think this can literally just be `ctx.next()`.

More generally, one *really cool* think about the middleware setup in this PR is that a lot of middleware won't have to allocate a new `FutureObj` at all!